### ISSUE TITLE
Controls the Placeholder to display only one line

### ISSUE THREAD

**Issue by JoakimLiu** [ruslanskorb/RSKPlaceholderTextView#28]
like UILabel lineBreakMode is byTruncatingTail, example: text: "abcdefghijklmnopqrstuvwxyz", only show: "abcdefghijklmno..."

**Comment by ruslanskorb** [comment]
@JoakimLiu Could you restate your thought?

**Comment by JoakimLiu** [comment]
Sorry
Placeholder displays only one line, if more than one line, truncated then the end show `...`.  

To the following UILabel example
```
let label = UILabel(frame: CGRect(origin: .zero, size: CGSize(width: 200, height: 36)))
label.font = font
label.numberOfLines = 1
label.lineBreakMode = .byTruncatingTail
label.text = "abcdefghijklmnopqrstuvwxyz"
```
Eventually, the UILabel will only display "abcdefghijklmno..."

**Comment by ruslanskorb** [comment]
For now you can achieve this by setting `attributedPlaceholder` to an attributed string with `paragraphStyle` that has `lineBreakMode` set to `.byTruncatingTail`.

```swift
textView.attributedPlaceholder = NSAttributedString(
    
    string: "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua.",
    attributes: [
        
        .font: textView.font ?? .systemFont(ofSize: UIFont.systemFontSize),
        .foregroundColor: textView.placeholderColor,
        .paragraphStyle: {
            
            let mutableParagraphStyle = NSMutableParagraphStyle()
            mutableParagraphStyle.alignment = textView.textAlignment
            mutableParagraphStyle.lineBreakMode = .byTruncatingTail
            return mutableParagraphStyle
        }()
    ]
)
```

**Comment by JoakimLiu** [comment]
Thanks, but it didn't  work for me, I rewrote my own placeholder textview.

![image](https://github.com/ruslanskorb/RSKPlaceholderTextView/assets/7654983/be964df4-7b75-4832-b795-ddfce191c626)

**Comment by ruslanskorb** [comment]
@JoakimLiu The reason for this is that you are specifying a foreground color other than `textView.placeholderColor` in the attributes.

When using the version of RSKPlaceholderTextView that is in the **master** branch, if you want to use a placeholder color other than the **default**, you first need to set it, and then [set the `attributedPlaceholder` that uses `textView.placeholderColor` as a foreground color](https://github.com/ruslanskorb/RSKPlaceholderTextView/issues/28#issuecomment-1872275895).

```swift
textView.placeholderColor = UIColor.gray
textView.attributedPlaceholder = ...
```

When using the version of RSKPlaceholderTextView that is in the **develop** branch, you can use any color directly in the attributes, as in your example.

Also, when using the version of RSKPlaceholderTextView located in the **develop** branch, to set a one-line limit for the placeholder, you just need to set the value of the `RSKPlaceholderTextView.placeholderLineBreakMode` property to `.byTruncatingTail`.

**Comment by JoakimLiu** [comment]
Thanks, for solving my problem. [In the code above](https://github.com/ruslanskorb/RSKPlaceholderTextView/issues/28#issuecomment-1882770793) (`master` branch), I only added this line of code
```
textView.placeholderColor = UIColor.gray
```

**Comment by ruslanskorb** [comment]
`placeholderLineBreakMode` is now available in the new [8.0.0](https://github.com/ruslanskorb/RSKPlaceholderTextView/releases/tag/8.0.0) version of `RSKPlaceholderTextView`.
